pyfifotax/data_structures_dataframe.py

```diff
@@ -0,0 +1,358 @@
+from __future__ import annotations
+
+from datetime import datetime
+from dataclasses import dataclass, asdict
+import pandas as pd
+from pandas.core.series import Series
+
+from typing import Optional
+
+
+@dataclass
+class DataFrameRow:
+    def to_dict(self) -> dict:
+        return asdict(self)
+
+    @staticmethod
+    def from_schwab_json(json_dict: dict) -> DataFrameRow:
+        raise NotImplementedError
+
+    @staticmethod
+    def empty_dict() -> dict:
+        raise NotImplementedError
+
+    @staticmethod
+    def from_df_row(row: Series) -> DataFrameRow:
+        raise NotImplementedError
+
+
+@dataclass
+class ESPPRow(DataFrameRow):
+    date: datetime
+    symbol: str
+    buy_price: pd.Float64Dtype
+    fair_market_value: pd.Float64Dtype
+    quantity: pd.Float64Dtype
+    currency: str
+
+    @staticmethod
+    def from_schwab_json(json_dict: dict) -> ESPPRow:
```

Comment:
It's not high priority, but I've just seen this now that I've started adapting IBKR converter. I think it's not the right place for this function to be.

ESPPRow (and other structures here) should not know about Schwab, IBKR, DeGiro, ING, DKB, etc. specifically. Instead, they should provide an interface (i.e. a constructor) to create them in a way the caller likes.

If the caller wants to do some JSON pre-processing, CSV parsing, binary decoding, dollar sign removal, locale conversion, etc. before getting the number "3" as the `buy_price`, that's on them, that process is uninteresting from ESPPRow standpoint.

In simpler terms, I recommend to move these functions into Schwab converter otherwise it's risking to create a [god class](https://en.wikipedia.org/wiki/God_object).

Reply:
Good point, for now the main advantage is that e.g. if we introduce a new column "product" or "comment", things are all in a single place. 
In the longer term, I agree that this pattern is not really the best. Let's address this later when addressing converters again in a more general fashion.

Reply:
Let's discuss this another time (not related to functionality, only code design)